helix-term/src/commands.rs

```diff
@@ -2021,8 +2021,9 @@ fn global_search(cx: &mut Context) {
 
         fn format(&self, current_path: &Self::Data) -> Row {
             let relative_path = helix_core::path::get_relative_path(&self.path)
-                .to_string_lossy()
-                .into_owned();
+                .map_or(SCRATCH_BUFFER_NAME.to_owned(), |path| {
+                    path.to_string_lossy().into_owned()
+                });
```

Comment:
Callers of `get_relative_path` should not use `SCRATCH_BUFFER_NAME` as a fallback value if they can't find the relative path. `SCRATCH_BUFFER_NAME` should only be used when a document doesn't have a path (i.e. is a "scratch buffer"). Let's use the absolute path as a fallback when `get_relative_path` returns an error as above